src/components/main-view/main-view.jsx

```diff
@@ -1,12 +1,25 @@
 import { useState, useEffect } from "react";
 import MovieCard from "../movie-card/movie-card";
 import MovieView from "../movie-view/movie-view";
+import LoginView from "../login-view/login-view";
+import SignupView from "../signup-view/signup-view";
 
 const MainView = () => {
+  const storedUser = JSON.parse(localStorage.getItem("user"));
+  const storedToken = JSON.parse(localStorage.getItem("token"));
   const [movies, setMovies] = useState([]);
+  const [selectedMovie, setSelectedMovie] = useState(null);
+  const [user, setUser] = useState(storedUser ? storedUser : null);
+  const [token, setToken] = useState(storedToken ? storedToken : null);
 
   useEffect(() => {
-    fetch('https://render-movie-api.onrender.com/movies')
+    //This ensures nothing is fetched when no token is present
+    if (!token) {
+      return;
+    }
+    fetch('https://render-movie-api.onrender.com/movies', {
+      headers: { Authorization: `Bearer ${token}` }
+    })
       .then((response) => response.json())
       .then((data) => {
```

Comment:
You should consider adding a catch statement to gracefully get cases where the token is expired or invalid.